src/source.cpp

```diff
@@ -0,0 +1,84 @@
+// Source.cpp
+// Central Source Class
+
+#include <iomanip>
+#include <string>
+#include <vector>
+
+#ifndef PYNE_IS_AMALGAMATED
+#include "particle.h"
+#include "source.h"
+#endif
+
+/***************************/
+/*** Protected Functions ***/
+/***************************/
+
+// there are no protected functions currently
+// fool.
+
+/************************/
+/*** Public Functions ***/
+/************************/
+
+/*--- Constructors ---*/
+
+pyne::Source::Source() {}
+
+// Destructor
+pyne::Source::~Source() {}
+
+std::ostream& operator<<(std::ostream& os, pyne::Source tal) {
+  // print the Source to ostream
+  os << "\t---------\n";
+  return os;
+}
+
+pyne::PointSource::PointSource(double x, double y, double z, double i, double j,
+                               double k, double E, std::string p, double weight)
+    : x(x), y(y), z(z), i(i), j(j), k(k), E(E), particle(p), weight(weight) {}
+
+// Destructor
+pyne::PointSource::~PointSource() {}
+
+std::string pyne::PointSource::mcnp(int version) {
+  std::stringstream output;  // output stream
+  
+  std::stringstream newline;
+  newline << std::endl << "     ";
+  
+  
+  output << "SDEF ";
+  if (x != 0 || y != 0 || z != 0) {
```

Comment:
I think we might as well write the information explicitly, even if it is the defaults.  No reason not to. The defaults are a convenience to the user in terms of keystrokes.   Writing out the explicit defaults will be a convenience for readers.

Reply:
I keep the explicit information for all except for the direction.
I felt like it make more sense to not indicate the default (0 0 0) for an isotropic source.
let me know if you disagree